src/autowiring/test/PostConstructTest.cpp

```diff
@@ -156,6 +198,15 @@ TEST_F(PostConstructTest, MultiNotifyWhenAutowired) {
   ASSERT_EQ(10, field) << "Autowiring lambdas did not run the expected number of times";
 }
 
+/*
+ Compile time failure case:
+class ForwardDeclared;
+TEST_F(PostConstructTest, TestForwardDeclare) {
+  Autowired<ForwardDeclared> decl;
+  decl.NotifyWhenAutowired([]() {std::cout << "thing"; });
+}
+*/
```

Comment:
No - I'm not sure how we want to handle the documentation of compile time failures? I'd like to leave it in